main.go

```diff
@@ -92,10 +95,114 @@ func startAccepting(maxClients int64, connType, connHost, connPort string, inter
 				LocalPort:  connPort,
 			}
 			clients <- c
+			go reportIPToAbuseIPDB(remoteIpAddr, records, abuseipdbeEnabled, abuseIpdbApiKey)
 		}
 	}()
 }
 
+func reportIPToAbuseIPDB(ip string, records chan<- metrics.RecordEntry, abuseipdbeEnabled bool, abuseIpdbApiKey string) {
+	if !abuseipdbeEnabled {
+		return
+	}
+	if isCached(ip) {
+		glog.V(1).Infof("IP is already cached, skipping report")
+		records <- metrics.RecordEntry{
+			RecordType: metrics.RecordEntryTypeReport,
+			IpAddr:     ip,
+			Message:    "IP is already cached, skipping report",
+		}
+		return
+	}
+
+	var apiKey string
+	abuseIpdbApiKeyFile := "ABUSE_IPDB_API_KEY_FILE"
+	if abuseIpdbApiKey != "" {
+		apiKey = abuseIpdbApiKey
+	} else if abuseIpdbApiKeyFile != "" {
+		key, err := os.ReadFile(abuseIpdbApiKeyFile)
+		if err != nil {
+			glog.V(1).Infof("Error reading API key file: %v", err)
+			records <- metrics.RecordEntry{
+				RecordType: metrics.RecordEntryTypeReport,
+				IpAddr:     ip,
+				Message:    fmt.Sprintf("Error reading API key file: %v", err),
+			}
+			return
+		}
+		apiKey = strings.TrimSpace(string(key))
+	} else {
+		glog.V(1).Infof("AbuseIPDB API key not set, skipping report")
+		records <- metrics.RecordEntry{
+			RecordType: metrics.RecordEntryTypeReport,
+			IpAddr:     ip,
+			Message:    "AbuseIPDB API key not set, skipping report",
+		}
+		return
+	}
+
+	reportURL := fmt.Sprintf("https://api.abuseipdb.com/api/v2/report?ip=%s&categories=18,22&comment=SSH login attempts with user root.&timestamp=2023-10-18T11:25:11-04:00", ip)
```

Comment:
Why is the timestamp static in here?

Reply:
another oversight, when debugging and forgot - will work on this on the weekend due to uni coursework - and most of the rest